jdbc-adapter/virtualschema-jdbc-adapter/src/main/java/com/exasol/adapter/dialects/oracle/OracleSqlDialect.java

```diff
@@ -178,12 +179,51 @@ public String generatePushdownSql(final ConnectionInformation connectionInformat
      * @return import type
      */
     public ImportType getImportType() {
-        if (this.properties.isEnabled(LOCAL_IMPORT_PROPERTY)) {
+        if (this.properties.isEnabled(IS_LOCAL_PROPERTY)) {
             return ImportType.LOCAL;
         } else if (this.properties.isEnabled(ORACLE_IMPORT_PROPERTY)) {
             return ImportType.ORA;
         } else {
             return ImportType.JDBC;
         }
     }
+
+    @Override
+    public void validateProperties() throws PropertyValidationException {
+        checkImportPropertyConsistency();
+        checkOracleSpecificPropertyConsistency();
+        super.validateProperties();
+    }
+
+    private void checkOracleSpecificPropertyConsistency() throws PropertyValidationException {
+        if (this.properties.containsKey(PROP_ORACLE_CAST_NUMBER_TO_DECIMAL)) {
+            final String dialectName = this.properties.getSqlDialect().toUpperCase();
+            if (!dialectName.equals(VALUE_SQL_DIALECT_ORACLE)) {
```

Comment:
You are in the Oracle dialect already so that check makes no sense.

Reply:
fixed